addOns/exportreport/src/main/java/org/zaproxy/zap/extension/exportreport/ExportReportAPI.java

```diff
@@ -276,6 +282,35 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
                                     "exportreport.message.console.info.pass.generate"));
                 }
 
+                int scanId = this.getParam(params, ACTION_PARAM_SCAN_ID, -1);
+
+                // check if scanid is valid when specified
+                ActiveScan scan = null;
+
+                if (scanId != -1) {
+                    ExtensionActiveScan extension =
+                            Control.getSingleton()
+                                    .getExtensionLoader()
+                                    .getExtension(ExtensionActiveScan.class);
+
+                    if (extension == null) {
+                        throw new IllegalArgumentException(
```

Comment:
It should use `ApiException` instead, e.g. `ApiException(ApiException.Type.BAD_STATE, "Active Scan extension not enabled.")`.

Reply:
Changed in the latest commit, but wanted to check: is it better to get the exception error from the external resource file or use a string literal?

Reply:
It's fine with a string literal.